### PR TITLE
Davidlee0602/rating

### PR DESCRIPTION
**Pull request by davidlee0602**
rating system implemented.

### CODE REVIEW

File: server/quizzes/routes.js [hunk 1]
```diff
@@ -28,6 +30,17 @@ quizRouter.get('/:id', async (req, res) => {
 //   .catch(error => res.status(500).send(error));
 // });
 
+// Send Score - will change to POST method.
+quizRouter.get('/:user/:id/:score', async (req, res) => {
```
Comment (jlin88) [inline server/quizzes/routes.js:34]: What is this route trying to do? Is this trying to get the user's score for a given quiz they have taken?

Comment (davidlee0602) [reply server/quizzes/routes.js:34]: Hi, it is the number of stars (out of 5) selected, indicating how they felt about the quiz. (poor thru excellent)
I am assuming the naming convention used here is a bit confusing. Should I change it to "rating" instead?

File: client/screens/Browse.jsx [hunk 2]
```diff
@@ -12,9 +12,13 @@ export const Browse = () => {
...
-    api.get(`/quizzes?search=${searchVal}&category=${categoryVal}`).then((res) => {
-      setQuizList(res.data);
-    });
+    // input delay to reduce number of
+    const delaySearch = setTimeout(() => {
+      api.get(`/quizzes?search=${searchVal}&category=${categoryVal}`).then((res) => {
+        setQuizList(res.data);
+      });
+    }, 200);
+    return () => clearTimeout(delaySearch);
```
Comment (dlimx) [inline client/screens/Browse.jsx:21]: Curious what this is for

File: client/screens/QuizRate.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const [rateVal, setRate] = useState(-1);
+  const api = useAPI();
+  const { id } = useParams();
+  const { currentUser } = useAuth();
+  // check login status?
+  // if (user) {
+  //     // User is signed in.
+  //     } else {
+  //     // No user is signed in.
+  //     }
```
Comment (dlimx) [inline client/screens/QuizRate.jsx:17]: I agree, maybe rate only for logged in users

File: client/screens/Browse.jsx (hunk 2 above)
Comment (davidlee0602) [reply client/screens/Browse.jsx:21]: This is to reduce the number of GET requests being sent to the server when a user types in the "Search Input Field" to filter/narrow their quiz search results. I've set a 200millisecond delay for throttling purposes.

File: server/quizzes/routes.js (hunk 1 above)
Comment (jlin88) [reply server/quizzes/routes.js:34]: Should this be a POST request? I think rating might be a better name. Perhaps the endpoint should be `users/:user_id/quizzes/:quiz_id/rating/:rating_id`

File: client/screens/Browse.jsx (hunk 2 above)
Comment (dlimx) [reply client/screens/Browse.jsx:21]: I haven't seen this before, nice!

**Review by dlimx — APPROVED** [review]

File: server/quizzes/routes.js (hunk 1 above)
Comment (davidlee0602) [reply server/quizzes/routes.js:34]: Yeah, it should be a POST request. I will make the modification in the coming days. I like the URL you have proposed. Think we should all have a discussion to get the route name to be somewhat uniform.

**Pull request merged**